### ISSUE TITLE
META | Implement automated code check?

### ISSUE THREAD

**Issue by MichaIng** [MichaIng/DietPi#1864]
As the "Checks" tab on PRs always pop into my eyes, I had a look into this code checker marketplace. For open source projects, everything I had a look at is for free.
E.g. `Codacy` supports shell and does code quality checks: https://github.com/marketplace/codacy
- Hmm seems to be the only tool that supports SHELL at all, if all the supported languages lists are complete 🤔.

Of course as we are legend, we don't need it 😀, but could be still interesting what they find?

**€: Travis CI has integrated ShellCheck (?): https://github.com/koalaman/shellcheck#travis-ci**

**Comment by Fourdee** [comment]
@MichaIng 

I tried codacy for a while, but didn't like it if i'am honest. No errors found, but its mostly "minor recommends" which do not suit our project coding style. eg:

```
if (( $var )); then
= $ not required. Yes I know, but I prefer it to match [[ $var ]] style code.
```

Couldn't find a way to change the detection options either, so removed it.

**Comment by MichaIng** [comment]
@Fourdee 
Looks like now this is possible:
- https://support.codacy.com/hc/en-us/articles/207994335-Code-Patterns
- https://support.codacy.com/hc/en-us/articles/208241349-Default-Patterns

**Comment by MichaIng** [comment]
Played a bid around:
- ~https://bettercodehub.com requires admin access, no cannot be used as contributor.~
  - **As I am repo owner now I could have a closer look. It just does very rough assumptions about script/block/loop sizes and such that mostly do not fit our coding style and needs.**
- https://www.codefactor.io allows to be used as contributor. But it identifies only files with `.sh`/`.bash` ending as shell scripts, so does not scan DietPi core scripts. I already opened an GitHub issue about that.
**EDIT: No joy so far: https://github.com/codefactor-io/codefactor/issues/2
EDIT2: Request has been added to their roadmap (see above), would be exceptional 😃!**
- https://app.codacy.com allows to customize file endings, but their support verified, that it is not possible to add empty/non file endings as well. He said he will forward the idea to allow that, we will see.

Finally, so far only PREP and bashrc.d/ scripts are scanned which makes the CI solutions more-less useless for DietPi currently 😉.
